react/features/follow-me/middleware.js

```diff
@@ -0,0 +1,177 @@
+// @flow
+
+import { MiddlewareRegistry } from '../base/redux';
+import {
+    getLocalParticipant,
+    getParticipantById,
+    getPinnedParticipant,
+    pinParticipant
+} from '../base/participants';
+import { CONFERENCE_WILL_JOIN } from '../base/conference';
+import { setFilmstripVisible } from '../filmstrip';
+import { setTileView } from '../video-layout';
+
+import { FOLLOW_COMMAND } from './constants';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+declare var APP: Object;
+
+/**
+ * The timeout after which a follow-me command that has been received will be
+ * ignored if not consumed.
+ *
+ * @type {number} in seconds
+ * @private
+ */
+const _FOLLOW_ME_RECEIVED_TIMEOUT = 30;
+
+/**
+ * An instance of a timeout used as a workaround when attempting to pin a
+ * non-existent particapant, which may be caused by participant join information
+ * not being received yet.
+ *
+ * @type {TimeoutID}
+ */
+let nextOnStageTimeout;
+
+/**
+ * A count of how many seconds the nextOnStageTimeout has ticked while waiting
+ * for a participant to be discovered that should be pinned. This varaible
+ * works in conjunction with {@code _FOLLOW_ME_RECEIVED_TIMEOUT} and
+ * {@code nextOnStageTimeout}.
+ *
+ * @type {number}
+ */
+let nextOnStageTimer = 0;
+
+/**
+ * Represents the &quot;Follow Me&quot; feature which enables a moderator to
+ * (partially) control the user experience/interface (e.g. filmstrip
+ * visibility) of (other) non-moderator particiapnts.
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case CONFERENCE_WILL_JOIN: {
+        const { conference } = action;
+
+        // Listen to "Follow Me" commands. I'm not sure whether a moderator can
+        // (in lib-jitsi-meet and/or Meet) become a non-moderator. If that's
+        // possible, then it may be easiest to always listen to commands. The
+        // listener will validate received commands before acting on them.
+        conference.addCommandListener(
+            FOLLOW_COMMAND, ({ attributes }, id) => {
+                _onFollowMeCommand(attributes, id, store);
```

Comment:
I don't usually give this type of feedback but I started to think about it now the 2nd or 3rd time since I'm reading the code, so I thought I just share my thoughts: do you think it would be better this command to be named ```FOLLOW_ME_COMMAND``` as the handler is also ```_onFollowMeCommand```.

Really just naming and I hate when others give this type of feedback, so feel free to ignore it, I just wanted to flush my thoughts here :)

Reply:
Consistency is important.